### PR TITLE
Adding a log event when reading .autorest_generated.json file

### PR DESCRIPTION
**Pull request by pragnagopa**
### Issue describing the changes in this PR

The change is to generate a log event when the `.autorest_generated.json` file exists in any existing function app. This file is added by stencil plugins and needs to be sent to inform us how many deployed function apps have been generated by Stencil plugins.

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script/Utility.cs [hunk 1]
```diff
@@ -792,6 +792,35 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
...
 
+        public static string ReadAutorestGeneratedJson(string rootScriptPath)
+        {
+            string autorestGeneratedJsonPath = Path.Combine(rootScriptPath, ScriptConstants.AutorestGeenratedMetadataFileName);
+            string autorestGeneratedJsonPathContents;
+            JObject autorestGeneratedJson;
+
+            try
+            {
+                autorestGeneratedJsonPathContents = FileUtility.ReadAllText(autorestGeneratedJsonPath);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: Check FileExists before trying to read - this would avoid try..catch

File: src/WebJobs.Script/Utility.cs [hunk 2]
```diff
@@ -792,6 +792,35 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
...
+            }
+
+            try
+            {
+                autorestGeneratedJson = JObject.Parse(autorestGeneratedJsonPathContents);
+            }
+            catch (JsonException ex)
+            {
+                // This is an error Stencil plugins should know about and not generate.
+                throw new FormatException($"Unable to parse autorest configuration file '{autorestGeneratedJsonPath}' with content '{autorestGeneratedJsonPathContents}'.", ex);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: Do not throw. Log error and return null

File: src/WebJobs.Script/Utility.cs [hunk 3]
```diff
@@ -792,6 +792,35 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
             }
         }
 
+        public static string ReadAutorestGeneratedJson(string rootScriptPath)
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs]: Pass in the logger as input - move the logic to log into this method. Rename this  method: `LogAutorestGeneratedJson`

Comment (pragnagopa) [reply src/WebJobs.Script/Utility.cs]: Moving the logic, will allow you to write unit tests - please add unit tests in `UtilityTests`

**Review by pragnagopa — COMMENTED** [review]
Added few comments. Thanks!

File: src/WebJobs.Script/Utility.cs [hunk 4]
```diff
@@ -792,6 +793,28 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
...
+            {
+                string autorestGeneratedJsonPathContents = FileUtility.ReadAllText(autorestGeneratedJsonPath);
+
+                try
+                {
+                    autorestGeneratedJson = JObject.Parse(autorestGeneratedJsonPathContents);
+                    logger.AutorestGeneratedFunctionApplication(autorestGeneratedJson.ToString());
+                }
+                catch (JsonException ex)
+                {
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs:810]: catch any `exception`  to ensure host continues to start if other exceptions happen here.

File: src/WebJobs.Script/Utility.cs [hunk 5]
```diff
@@ -792,6 +793,28 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
...
         }
 
+        public static void LogAutorestGeneratedJson(string rootScriptPath, ILogger logger = null)
+        {
+            string autorestGeneratedJsonPath = Path.Combine(rootScriptPath, ScriptConstants.AutorestGeenratedMetadataFileName);
+            JObject autorestGeneratedJson;
+
+            if (FileUtility.FileExists(autorestGeneratedJsonPath))
+            {
+                string autorestGeneratedJsonPathContents = FileUtility.ReadAllText(autorestGeneratedJsonPath);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs:803]: for adding unit tests, create a file in temp directory: https://stackoverflow.com/questions/278439/creating-a-temporary-directory-in-windows

and you can use `TestLogger` - you find references in other tests.

File: src/WebJobs.Script/Utility.cs [hunk 6]
```diff
@@ -792,6 +793,28 @@ public static bool IsMediaTypeOctetOrMultipart(MediaTypeHeaderValue mediaType)
             }
         }
 
+        public static void LogAutorestGeneratedJson(string rootScriptPath, ILogger logger = null)
+        {
```
Comment (pragnagopa) [inline src/WebJobs.Script/Utility.cs:797]: do not default to `null` - this will throw null ref if accidentally called without passing logger.

**Review by pragnagopa — COMMENTED** [review]
added few more comments

File: src/WebJobs.Script/Utility.cs (hunk 5 above)
Comment (vrdmr) [reply src/WebJobs.Script/Utility.cs:803]: Done. Thanks.

File: test/WebJobs.Script.Tests/UtilityTests.cs [hunk 7]
```diff
@@ -595,5 +600,74 @@ public void TryGetFunctionName_ReturnsExpectedResult(string keyName, bool expect
...
+  ""version"": ""0.2.0-preview""
+}
+";
+            string expected_contents = @"autorest_generated.json file found generated by Autorest (https://aka.ms/stencil) | file content";
+            File.WriteAllText(Path.Combine(functionAppDirectory, ScriptConstants.AutorestGeenratedMetadataFileName), sample_autorest_generated_json);
+            Utility.LogAutorestGeneratedJson(functionAppDirectory, _testLogger);
+            var allLogs = _testLogger.GetLogMessages();
+            VerifyLogLevel(allLogs, expected_contents, LogLevel.Information);
+        }
+
```
Comment (pragnagopa) [inline test/WebJobs.Script.Tests/UtilityTests.cs:642]: combine all these tests using `[Theory]` taking input context and expect log as input.

**Review by pragnagopa — COMMENTED** [review]
For the CI build to kick off - you need to create a branch on this repo directly. Please close this PR and create a new PR from a branch on this repo. Thanks!

**Comment by vrdmr** [comment]
Closing in favor of #6746

**Pull request closed without merging**